test/test_integration_single.rb

```diff
@@ -112,4 +112,23 @@ def test_siginfo_thread_print
 
     assert_match "Thread: TID", output.join
   end
+
+  def test_write_to_log
```

Comment:
Most of this is copied from `test/shell/t1.rb`. One of the main modifications was to remove the `#sleep` calls. I felt they were unnecessary now given the helper methods `#cli_server`, and `#stop-server` wait until the Puma starts fully before handing control back to the test.